src/super_gradients/training/models/sg_module.py

```diff
@@ -62,3 +63,10 @@ class to implement.
         """
 
         raise NotImplementedError
+
+    def predict(self, images, *args, **kwargs) -> Result:
+        raise NotImplementedError(f"`predict` is not implemented for {self.__class__.__name__}.")
+
+    def set_dataset_processing_params(self, *args, **kwargs) -> None:
+        """Set the processing parameters for the dataset."""
+        pass
```

Comment:
@shaydeci 
Not a big fan of having `pass` as default.
BUT with the design we discussed we want to apply `set_dataset_processing_params` to every dataset (see `net. set_dataset_processing_params=...` in `models.get`)
Because of that, having an abstract class or Raising `NotImplementedError` would make all other models crash...

An alternative solution could be to simply implement a default `set_dataset_processing_params` behavior (with `class_names` and `image_processor`)
OR to raise `NotImplementedError` and do try/except, but it looks even worse imo